pkg/types/config/context.go

```diff
@@ -72,6 +75,14 @@ type ScaledContext struct {
 	managementContext *ManagementContext
 }
 
+type CatalogManager interface {
+	ValidateChartCompatibility(template *managementv3.CatalogTemplateVersion, clusterName string) error
+	ValidateKubeVersion(template *managementv3.CatalogTemplateVersion, clusterName string) error
+	ValidateRancherVersion(template *managementv3.CatalogTemplateVersion) error
+	LatestAvailableTemplateVersion(template *managementv3.CatalogTemplate, clusterName string) (*v32.TemplateVersionSpec, error)
+	GetSystemAppCatalogID(templateVersionID, clusterName string) (string, error)
+}
+
```

Comment:
This should be in its own package or under catalog, it doesn't belong in context